libs/community/langchain_community/vectorstores/chroma.py

```diff
@@ -33,21 +33,34 @@
 DEFAULT_K = 4  # Number of Documents to return.
 
 
-def _results_to_docs(results: Any) -> List[Document]:
-    return [doc for doc, _ in _results_to_docs_and_scores(results)]
-
-
-def _results_to_docs_and_scores(results: Any) -> List[Tuple[Document, float]]:
-    return [
-        # TODO: Chroma can do batch querying,
-        # we shouldn't hard code to the 1st result
-        (Document(page_content=result[0], metadata=result[1] or {}), result[2])
-        for result in zip(
-            results["documents"][0],
-            results["metadatas"][0],
-            results["distances"][0],
-        )
-    ]
+def _results_to_docs(
+    results: Any, include_id: Optional[bool] = False
+) -> List[Document]:
+    return [doc for doc, _ in _results_to_docs_and_scores(results, include_id)]
+
+
+def _results_to_docs_and_scores(
+    results: Any, include_id: Optional[bool] = False
+) -> List[Tuple[Document, float]]:
+    # TODO: Chroma can do batch querying,
+    # we shouldn't hard code to the 1st result
+    output = []
+
+    for result in zip(
+        results["documents"][0],
+        results["metadatas"][0],
+        results["distances"][0],
+        results["ids"][0],
+    ):
+        metadata = result[1] or {}
+        if include_id:
+            metadata["id"] = result[3]
+
+        doc = Document(page_content=result[0], metadata=metadata)
+
+        output.append((doc, result[2]))
+
+    return output
```

Comment:
**Performance Issue**: The refactored methods '_results_to_docs' and '_results_to_docs_and_scores' now include an optional 'include_id' parameter which affects how metadata is handled. This change introduces additional processing for each document result to conditionally include an 'id' in the metadata if 'include_id' is set to True. This conditional check and potential dictionary modification inside a loop can lead to performance degradation, especially noticeable when processing large volumes of data. <br> **Fix**: To optimize the handling of 'include_id', consider preprocessing the results to separate those that require IDs from those that do not before entering the loop. This way, the conditional check is moved outside of the loop, reducing the computational overhead during the loop execution. <br> **Code Suggestion**: 
 ```
 +    if include_id:
 +        for result in zip(
 +            results["documents"][0],
 +            results["metadatas"][0],
 +            results["distances"][0],
 +            results["ids"][0],
 +        ):
 +            metadata = result[1] or {}
 +            metadata["id"] = result[3]
 +            doc = Document(page_content=result[0], metadata=metadata)
 +            output.append((doc, result[2]))
 +    else:
 +        for result in zip(
 +            results["documents"][0],
 +            results["metadatas"][0],
 +            results["distances"][0],
 +        ):
 +            metadata = result[1] or {}
 +            doc = Document(page_content=result[0], metadata=metadata)
 +            output.append((doc, result[2]))
 ```